optimum/utils/input_generators.py

```diff
@@ -575,3 +571,21 @@ def generate(self, input_name: str, framework: str = "pt"):
                 max_value=1,
                 framework=framework,
             )
+
+
+class DummyTrainingLabelsInputGenerator(DummyTextInputGenerator):
+    SUPPORTED_INPUT_NAMES = ("labels", "start_positions", "end_positions")
+
+    def generate(self, input_name: str, framework: str = "pt"):
+        max_value = 1 if self.task != "seq2seq-lm" else self.vocab_size
+        shape = [self.batch_size, self.sequence_length]
+        if self.task in [
+            "default",
+            "sequence-classification",
+            "multiple-choice",
+            "question-answering",
+            "image-classification",
+        ]:
+            shape = [self.batch_size]
```

Comment:
I prefer to check the label shape info in the mapping `_tasks_to_extra_inputs`, otherwise we need to update both in case of new tasks.

But as it is currently in the onnx config, there would be some circular import issue. I think that we should put it somewhere else, and for future shape info of other custom inputs, we can put all that here as well.

Reply:
I am not too sure about using the dynamic axes in the input generator. Dynamic axes values are quite specific to ONNX while `DummyInputGenerator` could be used for other exporters. Plus there could be other inputs with ndim > dynamic axes. WDYT?